src/Symfony/Component/Serializer/Normalizer/ObjectNormalizer.php

```diff
@@ -126,16 +140,7 @@ protected function extractAttributes($object, $format = null, array $context = [
      */
     protected function getAttributeValue($object, $attribute, $format = null, array $context = [])
     {
-        $cacheKey = \get_class($object);
-        if (!\array_key_exists($cacheKey, $this->discriminatorCache)) {
-            $this->discriminatorCache[$cacheKey] = null;
-            if (null !== $this->classDiscriminatorResolver) {
-                $mapping = $this->classDiscriminatorResolver->getMappingForMappedObject($object);
-                $this->discriminatorCache[$cacheKey] = null === $mapping ? null : $mapping->getTypeProperty();
-            }
-        }
-
-        return $attribute === $this->discriminatorCache[$cacheKey] ? $this->classDiscriminatorResolver->getTypeForMappedObject($object) : $this->propertyAccessor->getValue($object, $attribute);
+        return $this->propertyAccessor->getValue($object, $attribute);
```

Comment:
This looks like a BC break.

Reply:
Sorry, could you explain Why does it backward compatibility break ?

Reply:
@dunglas do you have any update on this for me ?

Reply:
@dunglas ping

Reply:
@dunglas , @nicolas-grekas Let's try to make a decision. What will we do next ? 
There are two options: 
1. We will fix this issue together. It means that I am waiting your propose how we fix it because my propose contains into this PR.
2. I'll close this issue and bug because of "looks like a BC break"